### PR TITLE
Let users pass any flag to Flux bootstrap (BREAKING see issue desc)

### PR DESCRIPTION
**Pull request by Callisto13**
### Description

Reverts the revert of https://github.com/weaveworks/eksctl/pull/3662, see that original PR for details.

Fix for issues has been noted below, I am running integration tests from this branch.

### Checklist
- [x] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [x] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes
- [x] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: pkg/flux/flux.go [hunk 1]
```diff
@@ -73,25 +57,33 @@ func (c *Client) runFluxCmd(args ...string) error {
...
-	if tokenPath == "" {
-		return executor.EnvVars{}, nil
+func (c *Client) checkFluxVersion() error {
+	logger.Debug(fmt.Sprintf("checking flux version is %s", minSupportedVersion))
+	out, err := c.executor.ExecWithOut(fluxBin, "--version")
+	if err != nil {
+		return err
+	}
+
+	trimmed := strings.TrimRight(string(out), "\n")
```
Comment (Callisto13) [inline pkg/flux/flux.go:67]: this fixes the issue with the integration tests. units have been updated

File: pkg/flux/flux.go [hunk 2]
```diff
@@ -37,32 +35,25 @@ func (c *Client) PreFlight() error {
...
+		return err
+	}
 
-	if c.opts.Kubeconfig != "" {
-		args = append(args, "--kubeconfig", c.opts.Kubeconfig)
+	args := []string{"check", "--pre"}
+	for k, v := range c.opts.Flags {
+		if k == "kubeconfig" {
+			args = append(args, fmt.Sprintf("--%s", k), v)
+		}
```
Comment (Callisto13) [inline pkg/flux/flux.go:46]: this fixes another issue with integration tests. units have been updated

Comment (aclevername) [reply pkg/flux/flux.go:46]: so only the kubeconfig flag gets passed to pre-flight? To we have to be wary about any other flags?

**Review by aclevername — APPROVED** [review]
LGTM, 1 comment open

File: Makefile [hunk 3]
```diff
@@ -134,14 +136,15 @@ integration-test-container: ## Run integration tests in a local container
 	  --env=AWS_SESSION_TOKEN \
 	  --env=AWS_ACCESS_KEY_ID \
 	  --env=AWS_SECRET_ACCESS_KEY \
+	  --env=GITHUB_TOKEN \
```
Comment (aclevername) [inline Makefile:139]: what did we use before? :thinking:

Comment (Callisto13) [reply Makefile:139]: the old flux tests did not need this

File: pkg/flux/flux.go (hunk 2 above)
Comment (aclevername) [reply pkg/flux/flux.go:46]: :+1: sounds reasonable

**Pull request merged**